res/themes/light-custom/css/_custom.scss

```diff
@@ -90,37 +92,73 @@ $roomtile-selected-color: var(--roomlist-text-color);
+$avatar-bg-color: var(--timeline-background-color);
+$button-secondary-bg-color: var(--timeline-background-color);
+$droptarget-bg-color: var(--timeline-background-color-50pct);
+$field-focused-label-bg-color: var(--timeline-background-color);
+$lightbox-border-color: var(--timeline-background-color);
+$menu-bg-color: var(--timeline-background-color);
+$message-action-bar-bg-color: var(--timeline-background-color);
+$primary-bg-color: var(--timeline-background-color);
+$reaction-row-button-bg-color: var(--timeline-background-color);
+$reaction-row-button-selected-border-color: var(--timeline-background-color);
+$roomtile-focused-bg-color: var(--timeline-background-color);
+$togglesw-ball-color: var(--timeline-background-color);
+$tooltip-timeline-fg-color: var(--timeline-background-color);
+//
+// --timeline-text-color
+$authpage-lang-color: var(--timeline-text-color);
+$dialog-title-fg-color: var(--timeline-text-color);
+$message-action-bar-fg-color: var(--timeline-text-color);
+$message-action-bar-hover-border-color: var(--timeline-text-color);
+$primary-fg-color: var(--timeline-text-color);
+$roomheader-color: var(--timeline-text-color);
+$roomtopic-color: var(--timeline-text-color-50pct);
+$settings-profile-overlay-placeholder-fg-color: var(--timeline-text-color);
+$tab-label-fg-color: var(--timeline-text-color);
+$tab-label-icon-bg-color: var(--timeline-text-color);
+$topleftmenu-color: var(--timeline-text-color);
+$tab-label-fg-color: var(--timeline-text-color);
 //
 // --timeline-text-secondary-color
 $authpage-secondary-color: var(--timeline-text-secondary-color);
-$memberstatus-placeholder-color: var(--timeline-text-secondary-color);
-$notice-secondary-color: var(--timeline-text-secondary-color);
-$pinned-color: var(--timeline-text-secondary-color);
-$settings-subsection-fg-color: var(--timeline-text-secondary-color);
-$roomheader-addroom-bg-color: var(--timeline-text-secondary-color);
-// was #747474
-$light-fg-color: var(--timeline-text-secondary-color);
-// was #777777
 $blockquote-fg-color: var(--timeline-text-secondary-color);
-// was #888888
+$event-timestamp-color: var(--timeline-text-secondary-color);
 $greyed-fg-color: var(--timeline-text-secondary-color);
 $info-plinth-fg-color: var(--timeline-text-secondary-color);
+$light-fg-color: var(--timeline-text-secondary-color);
+$memberstatus-placeholder-color: var(--timeline-text-secondary-color);
+$notice-secondary-color: var(--timeline-text-secondary-color);
+$pinned-color: var(--timeline-text-secondary-color);
 $preview-widget-fg-color: var(--timeline-text-secondary-color);
+$roomheader-addroom-bg-color: var(--timeline-text-secondary-color);
+$settings-subsection-fg-color: var(--timeline-text-secondary-color);
+$tooltip-timeline-bg-color: var(--timeline-text-secondary-color);
 //
-// --primary-color
-$accent-color-alt: var(--primary-color);
-$input-focused-border-color: var(--primary-color);
+// --username-color
+$username-variant1-color: var(--username-color);
```

Comment:
Here and in a few other places, you've added new variables to be set by the custom theme. Was it the effect for older custom themes that don't have a value `username-color` set?

Reply:
The effect for older custom themes which don’t set this css variable is … none, I’m pretty sure? If the default theme is used, the custom variable are ignore all together. If a custom theme is used, then only the re-defined (in `config.json`) variables are changed, all non-re-defined variables just fall back to their default values. Right?

Reply:
> If a custom theme is used, then only the defined custom variables (in `config.json`) are changed, all non-defined variables just fall back to their default values. Right?

I don't think this is true. SASS definitiions like `$username-variant1-color` are compile-time, and they can't fall back on previous definitions of the same variable depending on the runtime value of a CSS variable.

Reply:
I wonder if `$username-variant1-color: var(--username-color, $username-variant1-color);` would work though, using the default value argument in `var` to refer to the previous value of the SASS variable.

Reply:
> I wonder if `$username-variant1-color: var(--username-color, $username-variant1-color);` would work though, using the default value argument in `var` to refer to the previous value of the SASS variable.

Having carefully read through [the specs of CSS `var()`](https://developer.mozilla.org/en-US/docs/Web/CSS/var), I think you’re pretty right with this proposal.

Do you have a quick way to absolutely validate this?